src/Network/Ethereum/Web3/Types.hs

```diff
@@ -102,16 +124,19 @@ $(deriveJSON (defaultOptions
 data Call = Call
   { callFrom     :: !(Maybe Address)
   , callTo       :: !Address
-  , callGas      :: !(Maybe Text)
-  , callGasPrice:: !(Maybe Text)
-  , callValue    :: !(Maybe Text)
+  , callGas      :: !(Maybe Quantity)
+  , callGasPrice :: !(Maybe Quantity)
+  , callValue    :: !(Maybe Quantity) -- expressed in wei
```

Comment:
So, it's possible to add [Unit](http://hackage.haskell.org/package/web3-0.6.0.0/docs/Network-Ethereum-Unit.html#t:Unit) instance for `Quantity` to easy conversion from `Wei`, `Ether`, etc.

Reply:
I just added this function to make this easy.

```haskell
 setCallValue :: Unit a => a -> Call -> Call
 setCallValue value call = call { callValue = Just . Quantity $ toWei value }
```

Reply:
In my opinion, we don’t need special setter for this. For example, if `Quantity` will have `Unit` instance (`Unit` instance for `Maybe a` can be also created) we can write simple

```haskell
def { callValue = convert valueInEther }
```

OR with lens (may be its good practice for the future)

```haskell
callValue ~. convert valueInEther & def
```

---

``` haskell
instance Unit Quantity where
    fromWei = Quantity
    toWei = unQuantity

instance Unit a => Unit (Maybe a) where
    fromWei = pure . fromWei
    toWei Nothing = 0
    toWei (Just u) = toWei u
```

Reply:
I looked into making `Quantity` a `Unit`, but I get errors like

```
No instance for (Network.Ethereum.Unit.UnitSpec Quantity)
No instance for (Fractional Quantity)
```

I think there is a lot of boilerplate need to make `Unit` instances for not too much return. `Unit` is quite complicated and requires typeclasses like `Fractional` though I'm not sure why. I think a few simple functions like `fromEther`, `fromSzabo`, etc. would suffice. What do you think?

Reply:
So, `Fractional` is a simplest way to work with `Unit` derived types like a numbers, for example:

```
ghci> let x = 1 :: Quantity
```

Some words about `UnitSpec`, it's generalized way to describe ethereum unit value by two params: metric system scale (eg. 10^18 for ether) and text name.

For example, UnitSpec for Quantity is

```haskell
instance UnitSpec Quantity where
    divider = const 1
    name = const "quantity"
```

So, UnitSpec isn't exported from Network.Ethereum.Unit, it can be added to export.

Reply:
But an `Integer` (what a `Quantity` really is) is not fractional. `Wei` are not fractional. So I would have to write a non-sensical `Fractional` instance for `Quantity` to make this work. I'm going to try that right now to see what happens.

Reply:
You are really right, but generally Unit can be fractional, like `Ether`.
But `Wei` really rounded to `Integer` because `divider` of it is 1. Seems like fixed point math.

Reply:
Ok, I removed my `setCallValue` function and made `Quantity` a `Unit`. I think the `Fractional` instance I wrote for Quantity is OK.